x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -196,6 +220,19 @@ export const useColumns = ({
     setColumnsAndSave(populatedDefaultColumns);
   }, [browserFields, defaultColumns, setColumnsAndSave]);
 
+  /*
+   * In some case such security, we need some special fields such as threat.enrichments which are
+   * not fetched when passing only EMPTY_FIELDS. Hence, we will fetch all the fields that user has added to the table.
+   *
+   * Additionaly, system such as o11y needs fields which are not even added in the table such as rule_type_id and hence we
+   * additionly pass EMPTY_FIELDS so that it brings all fields apart from special fields
+   *
+   * */
+  const fieldsToFetch = useMemo(
+    () => [...columns.map((col) => ({ field: col.id, include_unmapped: true })), ...EMPTY_FIELDS],
```

Comment:
Because of performance issues It would be better if we could calculate this initially for the default columns and then once and then only for the new column that is added. Instead of running throw all of them every time, more so knowing that there has been already performance issues here and that columns.length can be thousands

Reply:
I am wondering if we should follow up with another PR to add a new prop to allow the fetch of extra fields if needed it. it might be cleaner? what do you think?

Reply:
100% agreed.